src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java

```diff
@@ -70,6 +87,48 @@
 
         providers.add(provider);
       }
+
+      if (generatorConfig.enableTestGenerator()) {
+        GapicCodePathMapper javaTestPathMapper =
+            CommonGapicCodePathMapper.newBuilder()
+                .setPrefix("src/test/java")
+                .setShouldAppendPackage(true)
+                .build();
+        GapicProvider<? extends Object> testProvider =
+            ViewModelDiscoGapicProvider.newBuilder()
+                .setModel(document)
+                .setProductConfig(productConfig)
+                .setSnippetSetRunner(new CommonSnippetSetRunner(new CommonRenderingUtil()))
+                .setModelToViewTransformer(
+                    new JavaSurfaceTestTransformer(
+                        javaTestPathMapper,
+                        new JavaDiscoGapicSurfaceTransformer(javaTestPathMapper, packageConfig),
+                        "java/http_unit_test.snip"))
+                .build();
+        providers.add(testProvider);
+      }
+      if (generatorConfig.enableSampleAppGenerator()) {
```

Comment:
@vam-google wants to remove sample app generation (just a note - I'm undecided yet on whether to remove it or not. Apparently it's broken for Gapic)

Reply:
Note taken.